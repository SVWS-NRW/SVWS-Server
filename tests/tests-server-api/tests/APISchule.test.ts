import {describe, expect, test} from "vitest";
import {getApiServer} from "./utils/TestUtils.js";

describe("Schule Tests ", () => {
	describe.each([{schema: "GymAbi01"}])('gegen %s', ({schema}) => {
		const api = getApiServer(schema);

		test("getSchullogo", async () => {
			const result = await api.getSchullogo(schema);
			expect(result).toMatchSnapshot();
		});

		test("getSchuleNummer", async () => {
			const result = await api.getSchuleNummer(schema);
			expect(result).toMatchSnapshot();
		});

		test("getSchuleStammdaten", async () => {
			const result = await api.getSchuleStammdaten(schema);
			expect(result).toMatchSnapshot();
		});

		test("getLernplattformen", async () => {
			const result = await api.getLernplattformen(schema);
			expect(result).toMatchSnapshot();
		});
	})
})
