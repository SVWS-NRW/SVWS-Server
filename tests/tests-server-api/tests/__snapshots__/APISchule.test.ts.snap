// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Schule Tests  > gegen { schema: 'GymAbi01' } > getSchuleNummer 1`] = `168890`;

exports[`Schule Tests  > gegen { schema: 'GymAbi01' } > getSchuleStammdaten 1`] = `
SchuleStammdaten {
  "abschnitte": ArrayList {
    "addFirst": [Function],
    "addLast": [Function],
    "elementData": [
      Schuljahresabschnitt {
        "abschnitt": 1,
        "id": 18,
        "idFolgeAbschnitt": 17,
        "idVorigerAbschnitt": null,
        "schuljahr": 2010,
      },
      Schuljahresabschnitt {
        "abschnitt": 2,
        "id": 17,
        "idFolgeAbschnitt": 16,
        "idVorigerAbschnitt": 18,
        "schuljahr": 2010,
      },
      Schuljahresabschnitt {
        "abschnitt": 1,
        "id": 16,
        "idFolgeAbschnitt": 6,
        "idVorigerAbschnitt": 17,
        "schuljahr": 2011,
      },
      Schuljahresabschnitt {
        "abschnitt": 2,
        "id": 6,
        "idFolgeAbschnitt": 15,
        "idVorigerAbschnitt": 16,
        "schuljahr": 2011,
      },
      Schuljahresabschnitt {
        "abschnitt": 1,
        "id": 15,
        "idFolgeAbschnitt": 13,
        "idVorigerAbschnitt": 6,
        "schuljahr": 2012,
      },
      Schuljahresabschnitt {
        "abschnitt": 2,
        "id": 13,
        "idFolgeAbschnitt": 9,
        "idVorigerAbschnitt": 15,
        "schuljahr": 2012,
      },
      Schuljahresabschnitt {
        "abschnitt": 1,
        "id": 9,
        "idFolgeAbschnitt": 14,
        "idVorigerAbschnitt": 13,
        "schuljahr": 2013,
      },
      Schuljahresabschnitt {
        "abschnitt": 2,
        "id": 14,
        "idFolgeAbschnitt": 12,
        "idVorigerAbschnitt": 9,
        "schuljahr": 2013,
      },
      Schuljahresabschnitt {
        "abschnitt": 1,
        "id": 12,
        "idFolgeAbschnitt": 2,
        "idVorigerAbschnitt": 14,
        "schuljahr": 2014,
      },
      Schuljahresabschnitt {
        "abschnitt": 2,
        "id": 2,
        "idFolgeAbschnitt": 8,
        "idVorigerAbschnitt": 12,
        "schuljahr": 2014,
      },
      Schuljahresabschnitt {
        "abschnitt": 1,
        "id": 8,
        "idFolgeAbschnitt": 7,
        "idVorigerAbschnitt": 2,
        "schuljahr": 2015,
      },
      Schuljahresabschnitt {
        "abschnitt": 2,
        "id": 7,
        "idFolgeAbschnitt": 11,
        "idVorigerAbschnitt": 8,
        "schuljahr": 2015,
      },
      Schuljahresabschnitt {
        "abschnitt": 1,
        "id": 11,
        "idFolgeAbschnitt": 10,
        "idVorigerAbschnitt": 7,
        "schuljahr": 2016,
      },
      Schuljahresabschnitt {
        "abschnitt": 2,
        "id": 10,
        "idFolgeAbschnitt": 3,
        "idVorigerAbschnitt": 11,
        "schuljahr": 2016,
      },
      Schuljahresabschnitt {
        "abschnitt": 1,
        "id": 3,
        "idFolgeAbschnitt": 4,
        "idVorigerAbschnitt": 10,
        "schuljahr": 2017,
      },
      Schuljahresabschnitt {
        "abschnitt": 2,
        "id": 4,
        "idFolgeAbschnitt": 5,
        "idVorigerAbschnitt": 3,
        "schuljahr": 2017,
      },
      Schuljahresabschnitt {
        "abschnitt": 1,
        "id": 5,
        "idFolgeAbschnitt": 1,
        "idVorigerAbschnitt": 4,
        "schuljahr": 2018,
      },
      Schuljahresabschnitt {
        "abschnitt": 2,
        "id": 1,
        "idFolgeAbschnitt": 19,
        "idVorigerAbschnitt": 5,
        "schuljahr": 2018,
      },
      Schuljahresabschnitt {
        "abschnitt": 1,
        "id": 19,
        "idFolgeAbschnitt": null,
        "idVorigerAbschnitt": 1,
        "schuljahr": 2019,
      },
    ],
    "getFirst": [Function],
    "getLast": [Function],
    "modCount": 19,
    "removeFirst": [Function],
    "removeLast": [Function],
  },
  "anzJGS_Jahr": 1,
  "bezeichnung1": "Albert Zweistein Gymnasium",
  "bezeichnung2": "Städt. Gymnasium für Jungen und Mädchen",
  "bezeichnung3": "- Sekundarstufen I und II -",
  "dauerUnterrichtseinheit": 45,
  "email": "zweistien@schule.nrw.de",
  "fax": "020212367",
  "hausnummer": "4",
  "hausnummerZusatz": "",
  "idSchuljahresabschnitt": 1,
  "ort": "Wuppertal",
  "plz": "42103",
  "schulNr": 168890,
  "schuleAbschnitte": SchuleAbschnitte {
    "abschnittBez": "Halbjahr",
    "anzahlAbschnitte": 2,
    "bezAbschnitte": ArrayList {
      "addFirst": [Function],
      "addLast": [Function],
      "elementData": [
        "1. Hj.",
        "2. Hj.",
      ],
      "getFirst": [Function],
      "getLast": [Function],
      "modCount": 2,
      "removeFirst": [Function],
      "removeLast": [Function],
    },
  },
  "schulform": "GY",
  "strassenname": "Viehofstraße",
  "telefon": "020255512345",
  "webAdresse": null,
}
`;

exports[`Schule Tests  > gegen { schema: 'GymAbi01' } > getSchullogo 1`] = `"/9j/4AAQSkZJRgABAQEASABIAAD/2wBDAAoHBwgHBgoICAgLCgoLDhgQDg0NDh0VFhEYIx8lJCIfIiEmKzcvJik0KSEiMEExNDk7Pj4+JS5ESUM8SDc9Pjv/2wBDAQoLCw4NDhwQEBw7KCIoOzs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozv/wAARCAB8AH8DASIAAhEBAxEB/8QAHwAAAQUBAQEBAQEAAAAAAAAAAAECAwQFBgcICQoL/8QAtRAAAgEDAwIEAwUFBAQAAAF9AQIDAAQRBRIhMUEGE1FhByJxFDKBkaEII0KxwRVS0fAkM2JyggkKFhcYGRolJicoKSo0NTY3ODk6Q0RFRkdISUpTVFVWV1hZWmNkZWZnaGlqc3R1dnd4eXqDhIWGh4iJipKTlJWWl5iZmqKjpKWmp6ipqrKztLW2t7i5usLDxMXGx8jJytLT1NXW19jZ2uHi4+Tl5ufo6erx8vP09fb3+Pn6/8QAHwEAAwEBAQEBAQEBAQAAAAAAAAECAwQFBgcICQoL/8QAtREAAgECBAQDBAcFBAQAAQJ3AAECAxEEBSExBhJBUQdhcRMiMoEIFEKRobHBCSMzUvAVYnLRChYkNOEl8RcYGRomJygpKjU2Nzg5OkNERUZHSElKU1RVVldYWVpjZGVmZ2hpanN0dXZ3eHl6goOEhYaHiImKkpOUlZaXmJmaoqOkpaanqKmqsrO0tba3uLm6wsPExcbHyMnK0tPU1dbX2Nna4uPk5ebn6Onq8vP09fb3+Pn6/9oADAMBAAIRAxEAPwD2R3WONpHOFUEk+gFcunxJ8MyoHiuLqRG5DLZTEH8dtdHe/wDHjcf9cm/lWF8O/wDkn+i5GP8ARh/M0AR/8LF8O9pL0/8AbhN/8TR/wsTw/jO6+I/68Jv/AImuoooA5b/hYugHp9v/APACb/4mj/hYmg9l1A/9uE3/AMTXU0UAcqfiLoY/5ZakfpYS/wDxNV7n4p+GbPb9qa+g3fd8yzkXP5iui13WbbQNIm1G6yyxgBY1+9I54VR7k8VxGueG5W8Ga14h18LPrM9k5VTylmmMiNB2x3PUnNMDdX4iaKyhlttUIIyCNPl5/Sl/4WFpJ6WWrn6adL/hXQ2H/IOtv+uS/wAhVikBy3/CwdKPSw1g/wDcOk/wpv8AwsPS+2m6yfpp8n+FdXRT0A5T/hYOnnppGuH6ac9L/wAJ/ZdtG13/AMFr1sReIdHm1FtOTUYPtiyNGYGba+5QCQAevBFaVGgHLReP9Oku7a2k03Vrc3MyQI9xZNGm5jgAk11B4HTNc540/wCPfR/+wzaf+jBXSUgILz/jxn/65t/KsH4dnPw/0X/r2H8zW/ef8eU//XNv5Vz/AMORj4f6MP8Ap3/qaAOmooooAKKKQkKCSQAOSTQBymop/bfxBsdPbJtdHg+2yr2aZiVjz9AGNXPHn/Iia10/485Ov0qv4Nb+0rrWfEA/1eo3ey3OPvQxDYp+hO41Y8ejPgPW/wDrzk/lTW4GvpxzplqfWFP5CrNZkOoWemeHrW7vrmK2gWCPMkrBQPlHrV62uYLy2jubWZJoZBuSSNgVYexpAYereKXttXGjaRprarqCp5k0azCJIVPTcx4yew61heIvEXjXStN+2z6fZ2cBba32dXu5Yx6kfKoHuTUXjnw1dXHinTtYsdMeeNCrXDWYQTs6kFcliOMDGa1U+JGj291LZ63FcaNdRqG8q6UNvU91KZB+lMDkrP4cahrmrQ65f3lwv2y4eSWZVFvOqbAY2KEfK28YwO3Oa1rLUPEXhDxZa6TrOpRXumanMwgubh2aSPHCoTgDJ47Y967XRdVXWtMS/S2nt4pC3lrOu1mUHhsdgetcv47gsk8S+F7+9tDLDHdNG8gTcAWA2A/8Dwc+1AGr40/49dI/7DNn/wCjRXR1znjQj7JpP/YZs/8A0aK6OgCG6GbSYesbfyrn/hz/AMk/0f8A69/6muin/wBRJ/un+Vc58OP+SfaP/wBcD/6EaQHT0UUUAFch4i1KfX75/CmiSkO3GpXadLWI9VB/vt0A7Ul1qmp+LNQm0zw/cG00y3fy7zVE+87DrHD7ju3bt79Do+i2Gg2C2WnwCKMHLEnLO3dmPUk+tAFiys7fT7KGztYxHBAgjjQdlAwK574k38Fh4B1Vp2x50BhjHdnbgD/PpXTO6RRtJIyoiglmY4AA6kmvL/FPmeNNI1fW2VhoumWkw08Hj7TLjDTf7o5C/nTQFlLzSpNXmvvEtyBp+h2tolvC6loxJLHkuVGdzdh6Crd2p8Cp/b+iutz4euGElzZhxiLd/wAtIT75GV/L25fWpRBDrM5j8wQ/2PNtK/KwCc7vQe9S3Nw2keJdI0dYZ9X0Eu2o6dBaL5jsCDtQgnlVbJGe1AHZfbfEvioY0+J9B0xv+Xqdc3Mo/wBhOiD3PNa+jeFtJ0NH+zW/mTynMtzOfMllPqzH/wDVVKy8W3t5eQwHwprMCSOFMs0aKqA9z83SuloAqapqVto+mXGo3jFLe3TfIVGSB9K8nttc1LU9dt/EGpWkN9ITLBpFiFZZPM35Vip4wF5L/wBa6m+tX8Y6nrttf6jdWul6Wfs/2W1YI0vyBmZzgkg5wB04qX4daNGmjweIrx2mvLyECN5W3fZ4B9xFJ9sEnuaAOO0DVLvU/tMmoXxubo+ILBmTBAjAl28DsMg4HoB617NXlNpdW+py3+pwSRypceJ7IxyZG9oxIoXI7Lndj15NerUMBk3+pf8A3T/Kub+G/wDyT/SP+uJ/9Caulk/1TfQ1zXw3/wCSf6R/1yP/AKG1IDp65jxfe3VxJaeGtMkMd5qmfMmU828C43v9T90e5rp65Xw0h1PxTruuucqko0+29kj5cj6uT+VAHQabp1rpOnQWFlEIreBAiKPT/Gn3t7badZy3l5MsMEKlpJHOAoqeuLlhPjjxLJHKSdA0ebaU/hu7kdc+qp+poAjW31L4gOs155un+G85jtvuzXw7F/7qH07/AK1r+M4Yrb4f6vDBGsccdhIqIowFAXgCugAAGAMAVheOefA2tf8AXlJ/6Caa3AwtT8OX8ukWutaAyi/OmpBcWz/cu4tn3T6MM8H8K5jQ77UfDEd68WnxyausenWMUdzx5ZdDwe4GR0zXqmhkHQdPI6fZYv8A0EV514psry51zxPHZJuu43sL2FQ4V9kYO51zwccihAavhrxN4v1y7MMsOiwSW0oS8tX81J4lzyQOQcjoeld7XDa+1jqfha38cWkkumX0FuJoJ1UM7A9InA+8CSBj3q7pnju2GnXH/CQRHStRsYfNubaT+Jcfej/vA9PY8UAP8R6Bex3r6/oODfbNl1aMcR3sYH3T6MB0Nc/4D8VW9lHF4fvSVs3kaKyllGGQ55t5R/C4zgdiK9CsLsahp9veLDLCs8YkEcq4dQRnBHY1wPxP8P2ttYy+IoIk3ErFewk7UnUnarHHR1JBDDmgDPs9KtLVr29s7aO2jfxPaWqQxjARYplGfqSxNerV5D4fttQXRheXZlaC51vTTHJIpXznWRFkkAPOGOOe+M169QwGuu5GX1GK5r4bn/igdKH91HU/UOw/pXT1xHhXWLDw74f1WHUblYY9M1OeLBPzEM29AB3JDcAUgOi8S6yNC0Wa7VfMuGxHbRDkyytwigfX9M0eGtJOh+H7Swdt8yKWnf8AvyMSzn/vomsnR7C91/WI/EmswNbxQgjTbF+sQPWVx/fI6DsK6ugDI8Vao2jeF9Rv4ziWKE+V/vn5V/8AHiKf4a0hNC8O2WnL96GIeY3dnPLE/Viaw/Ej/wDCReJ9P8MwHdBaut9qJHRVU5jQ+7Nzj0FdhQAVz3j5xH4C1ticf6G4/MYroa5P4k7pvCTadGf3upXMNon1Zxn9AaaA2vDkqzeGdLkU5DWkR/8AHBVLxR4Wi8QQpNBcNZalbqwt7uPqoI5Vh/Ep7iq/w7naTwXZ28pzNZM9rJ7NG5X+QFX/ABH4ig8P2sZMTXV5cN5draRffnf0HoPU9qSAy7nwncn4dQ+Hop42vLaCLy5D9xpI2DD8CRiuf8S6pp/iTw3BqV1YCG/0W+ha/t5F/e26bxv+qnr6ECta91DxnawC4vtS8M6WjEDZKXJUn+EsSAT9Kws2934mttY1bxh4cHlKY50tcD7TGRjY+WII+vSmB6hFJHNEksTh43UMrKcgg9CK5XxJv1jxbo/h5jstFU6hcf8ATXy2AVPpuOT9BXPaF4li8O6jdQ2JuL7wlFIEW82kpZu3VVPVox69s/nueNDLplzpvi+wmVjastvNGRuWaCV1HB7EHBBoAv8AjRgLPSh3bWbID/v8p/pXR1y3i4m51rwzpictJqIuWH+zCpYn8yK6mkAVw76RYWfxYW6u7VJP7TtfMtXcZCTx8PgdMlMHPXiu4rn/ABlpNzqOjrc6cB/aWnSi6tD6uvVfowyPxoA6Cue8S+J/7KaPTNNh+3a1dDFvaqfu/wC25/hUdeay5vG8uu21lY+FlV9Tv4vMkaQZWwTOGaT/AGgcgDua3dA8NWWgRO0bPc3s/NzezndLM3uew9AOBQA3wx4f/sGxk8+c3WoXb+deXJ6yyH0/2R0AraoooAK5TUiNZ+IOm6cvzQ6RE17P6eY3yRg++NzVvavqlroulXGpXj7IbdCzep9APcnisbwhZy6fpN1rWsFYr3UnN3dFzgRLj5Uz6Kv9aAIdCP8AY3jfWdGb5Yb/ABqNr6En5ZQPxAP41NrHgaz1nXBrD6pqlrdLH5am2uAgRe4HGRnvTfGdncNZ2mv6UvmX2kOZ41X/AJbREYkT8V/lW7pepWusaZb6jZyeZBcIHRvb0+o6UwMCx+HPh61uze3UMuqXRJJmv5DKWzxyDwcDpkVrr4b0FD8ui6eD7Wqf4Uuv67ZeHNHm1O+YiKIcKOrseij3NeZ3PiOw1qKK61fXNSubi8kxHpGhycQpg8NjljjqfwFAHTX9ynicz6LpbpZeHrTK6jfJhFkx1hjPTH95u3SnWtsfGktvsja28L2LL9niwQb5k+63qIxjj1xUdhoeoeILe0sbzShofh22AYaeJA0l0eoD4+6vcjqT1rofEmsx+HNBeeGINOcQWduo/wBZIeEUD0/oKAM2w/4nPxEvr4c2+j24s4j2Mr/NJj6DaK6usjwvop0LQobSR/MuWJlupe8krHLn8+PoBWvSAKKKKAOJu7ZfA3iO412CEf2NqjKNQ2rzaydpf9w559Cc12kbpLGskbq6OAyspyCD0INJLFHPC8MyLJHIpV0YZDA9QRXHGDUfAUhazhm1Dw4Tua3XLTWHqU7tH7dRQB2lNZlRC7sFVRkknAArMHifQzo39s/2pb/YMf67fxn0x1z7da53bqXxBYebHNpvhoHO1spPfj37rH+p/kAZOveJLXWJn1e5liGhaUWks4ZHCHU7hOhUHqin8zWFqcWv614fufEt1qLtYKpdhNO0Md2neKOJeiHpluTjtXW/EjTydL0bTbBFtU+0MsLxoAYnWJjGqntuYAVzdhD4e0e68OXGoXU99pV3YtILa4czpbz4GW8vHC/eHQ4NMRDbyeI/CumWesf2o01pJGCk63DzW4yOIZUP3f7odeh6itvQPEFvoLx6lA6Hw7qpWSWNHD/2XcP1DY6Ix/I1jX0Xh++i8UX9nNc29haCJrSxikaFXuWHDiL0LbccdjXU/DeyEmg6tpt9HFdwC9eN5WjH79iql93rhiRn2oGdwyQ3UIDrHNE4BAIDKw7VHHYWcMolitIEkHRljAP51x4XVPh8xWOOfVPDecgL809iP/Z0/Ufz3pfF/h+LR11b+04XtX4QodzO390L13e2M0gNS6urextZbq6mSGCJSzyOcBQO9cvoltP4o1lPFGoRNFZQArpVrIOQD1mYdmbt6Cmw6XqPjG6jvtet3s9JiYPbaW335iOjzf0T8/frwAoAAAA4AFAC0UUUAFFFFABRRRQBx2vfDjS9Tvk1TTiunajG/mhhEskTv6vGeCfcYqQeJPEOijZ4g8PyXES/8vulZlQ+5jPzL+tdbRQByV14k8G+KtOksZtVtyCQwjlkNvIjjkEbsEMD3FcBpWg694WvYNQtZ4h59sY5JbGEXpDb92HBYHJz1XjivYL3SNM1JSL7T7W6B/57Qq/8xWNJ8O/CEpJbQrYZ/ubl/kRTA89vfDmreI9UbV9Su7dERoF8+5QWU21Wy2wBjjAPVuT2xXdW/inwb4asItMsdQhkWEYSC1zO7Hqfu5ySeSTVqL4e+EYWDLoNqSP74L/zJrZs9NsNOXbY2Vvaqe0MSp/IUAc4db8Ua6Nmi6L/AGZbt/y+apw2PVYhz+ZxUvh3wDpGg3TahIovdSkYu9zKijax67EHC/hz711FFIAooooAKKKKAP/Z"`;
